src/pipelines/vaccinations/sk_authority.py

```diff
@@ -0,0 +1,136 @@
+            raise ConnectionError(f"Could not connect to {url}")
+        rows = res.json()
+    except Exception as exc:
+        if log_func:
+            log_func(requests.get(url_fmt, **get_opts).text)
+        raise exc
+
+    if offset is None:
+        # This is a non-recursive call.
+        return rows
+
+    if "next_offset" not in rows or rows["next_offset"] == None:
+        return rows["page"]
+    else:
+        return rows["page"] + _download_from_api(url, offset=rows["next_offset"])
+
+
+class SlovakiaDataSource(DataSource):
+    def fetch(
+        self,
+        output_folder: Path,
+        cache: Dict[str, str],
+        fetch_opts: List[Dict[str, Any]],
+        skip_existing: bool = False,
+    ) -> Dict[str, str]:
+
+        data_files = {}
+        for fetch_opt in fetch_opts:
+            # Base URL comes from fetch_opts
+            url_base = fetch_opt["url"]
+            name = fetch_opt["name"]
+
+            # Create a deterministic file name
+            file_path = (
+                output_folder
+                / "snapshot"
+                / ("%s.%s" % (uuid.uuid5(uuid.NAMESPACE_DNS, url_base), "json"))
+            )
+            # Avoid download if the file exists and flag is set
+            if not skip_existing or not file_path.exists():
+                with open(file_path, "w") as fd:
+                    if name == "regions":
+                        json.dump(_download_from_api(url_base), fd)
+                    elif name == "vaccine_by_regions":
+                        json.dump(_download_from_api(url_base, offset=0), fd)
+            data_files[name] = str(file_path.absolute())
+
+        return data_files
+
+    def parse_dataframes(
+        self, dataframes: Dict[Any, DataFrame], aux: Dict[str, DataFrame], **parse_opts
+    ) -> DataFrame:
+
+        if "regions" not in dataframes or "vaccine_by_regions" not in dataframes:
+            raise ArgumentError("Missing keys in dataframes dict.")
+
+        data = merge(
+            dataframes["vaccine_by_regions"],
+            dataframes["regions"],
+            left_on="region_id",
+            right_on="id",
+            how="left",
+        ).reindex()
+
+        print(data.columns)
+
+        data = table_rename(
+            data,
+            {
+                "title": "subregion1_name",
+                "published_on": "date",
+                "dose1_count": "new_persons_vaccinated",
+                "dose1_sum": "total_persons_vaccinated",
+                "dose2_count": "new_persons_fully_vaccinated",
+                "dose2_sum": "total_persons_fully_vaccinated",
+            },
+            drop=True,
+        )
+
+        data["new_vaccine_doses_administered"] = (
+            data["new_persons_vaccinated"] + data["new_persons_fully_vaccinated"]
+        )
+        data["total_vaccine_doses_administered"] = (
+            data["total_persons_vaccinated"] + data["total_persons_fully_vaccinated"]
+        )
+
+        data["key"] = None
```

Comment:
No need to set `key` as `None` unless some of the records have a non-null value.

Reply:
Done.